src/python_testing/TC_ACE_1_5.py

```diff
@@ -0,0 +1,164 @@
+        return current_fabric_index
+
+    async def write_acl(self, acl, th):
+        # This returns an attribute status
+        result = await th.WriteAttribute(self.dut_node_id, [(0, Clusters.AccessControl.Attributes.Acl(acl))])
+        asserts.assert_equal(result[0].Status, Status.Success, "ACL write failed")
+
+    async def read_descriptor_expect_success(self, th):
+        cluster = Clusters.Objects.Descriptor
+        attribute = Clusters.Descriptor.Attributes.DeviceTypeList
+        await self.read_single_attribute_check_success(
+            dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute)
+
+    async def read_basic_expect_success(self, th):
+        cluster = Clusters.Objects.BasicInformation
+        attribute = Clusters.BasicInformation.Attributes.VendorID
+        await self.read_single_attribute_check_success(
+            dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute)
+
+    async def read_basic_expect_unsupported_access(self, th):
+        cluster = Clusters.Objects.BasicInformation
+        attribute = Clusters.BasicInformation.Attributes.VendorID
+        await self.read_single_attribute_expect_error(
+            dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute, error=Status.UnsupportedAccess)
+
+    async def read_descriptor_expect_unsupported_access(self, th):
+        cluster = Clusters.Objects.Descriptor
+        attribute = Clusters.Descriptor.Attributes.DeviceTypeList
+        await self.read_single_attribute_expect_error(
+            dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute, error=Status.UnsupportedAccess)
+
+    @async_test_body
+    async def test_TC_ACE_1_5(self):
+        self.print_step(1, "Comissioning, already done ")
+        self.th1 = self.default_controller
+
+        self.print_step(2, "TH1 opens the commissioning window on the DUT")
+        params = self.OpenCommissioningWindow()
+
+        new_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        new_fabric_admin = new_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=self.matter_test_config.fabric_id + 1)
+
+        TH1_nodeid = self.matter_test_config.controller_node_id
+        TH2_nodeid = self.matter_test_config.controller_node_id + 2
+
+        self.th2 = new_fabric_admin.NewController(nodeId=TH2_nodeid,
+                                                  paaTrustStorePath=str(self.matter_test_config.paa_trust_store_path))
+
+        errcode = self.th2.CommissionOnNetwork(
+            nodeId=self.dut_node_id, setupPinCode=params.setupPinCode,
+            filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=self.matter_test_config.discriminators[0])
+        logging.info('Commissioning complete done. Successful? {}, errorcode = {}'.format(errcode.is_success, errcode))
+        self.print_step(3, "TH2 commissions DUT using admin node ID N2")
```

Comment:
can you move this to above the commissioning call? That way, the TH will correctly mark the failed step if it fails.

Reply:
Yes, I dit it